lib/bolt/inventory/target.rb

```diff
@@ -143,12 +143,28 @@ def port
         @uri_obj.port || transport_config['port']
       end
 
+      # For remote targets, protocol is the value of the URI scheme. For
+      # non-remote targets, there is no protocol.
       def protocol
-        transport
+        if remote?
+          @uri_obj.scheme
```

Comment:
This would be `@uri_obj.scheme || transport_config_cache['protocol']` if we were to expose a `protocol` config option. Then it would mirror the `transport` method and reinforce the idea that the URI scheme gets assigned to _either_ the protocol _or_ the transport depending on whether the target is remote.